services/power/client/client.go

```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2019 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+// Package client provides the client interface for 'power'
+package client
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/Snowflake-Labs/sansshell/client"
+	"github.com/google/subcommands"
+
+	pb "github.com/Snowflake-Labs/sansshell/services/power"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+)
+
+const subPackage = "power"
+
+func init() {
+	subcommands.Register(&powerCmd{}, subPackage)
+}
+
+type powerCmd struct{}
+
+func (*powerCmd) GetSubpackage(f *flag.FlagSet) *subcommands.Commander {
+	c := client.SetupSubpackage(subPackage, f)
+	c.Register(&rebootCmd{}, "")
+	return c
+}
+
+func (*powerCmd) Name() string { return subPackage }
+func (p *powerCmd) Synopsis() string {
+	return client.GenerateSynopsis(p.GetSubpackage(flag.NewFlagSet("", flag.ContinueOnError)), 2)
+}
+func (*powerCmd) Usage() string { return "" }
+
+func (*powerCmd) SetFlags(f *flag.FlagSet) {}
+
+func (p *powerCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	c := p.GetSubpackage(f)
+	return c.Execute(ctx, args...)
+}
+
+type rebootCmd struct {
+	when    string
+	message string
+}
+
+func (*rebootCmd) Name() string { return "reboot" }
+func (*rebootCmd) Synopsis() string {
+	return `reboot:
+  Reboot a host at a predefined time.
+ `
+}
+func (a *rebootCmd) Usage() string {
+	return client.GenerateUsage(subPackage, a.Synopsis())
+}
+
+func (a *rebootCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&a.when, "when", "now", "When to reboot a host. Valid values are 'now', 'hh:mm', '+5m'.")
```

Comment:
`+5m` won't work, so perhaps change this to "`+m` where m is the number of minutes "

Reply:
Thanks. I should not advertise what is obviously wrong